src/js/react/media-store.tsx

```diff
@@ -0,0 +1,116 @@
+import { createContext, useContext, useEffect, useMemo } from 'react';
+import type { Context, ReactNode } from 'react';
+import React from 'react';
+import { useSyncExternalStoreWithSelector } from './useSyncExternalStoreWithSelector';
+import createMediaStore from '../media-store/media-store';
+import type {
+  FullScreenElementStateOwner,
+  MediaState,
+  MediaStateOwner,
+  MediaStore,
+} from '../media-store/media-store';
+
+// The context is the thing that allows for a shared... context that propagates outside of the (V)DOM/Component tree.
+// Contexts have values. Our value will be the the media ui store (aka the "guts" of Media Controller state mgmt)
+export const MediaContext: Context<MediaStore | null> =
+  createContext<MediaStore | null>(null);
+
+// The Provider is a (non-visual) React Element that... provides a context for any descendant in the Component tree that may
+// want to use it. As such, it must be an ancestor in the Component Tree for anything that wants to "consume" the context.
+export const Provider = ({
+  children,
+  mediaStore,
+}: {
+  children: ReactNode;
+  mediaStore?: MediaStore;
+}) => {
+  const value = useMemo(
+    () =>
+      mediaStore ?? createMediaStore({ documentElement: globalThis.document }),
+    [mediaStore]
+  );
+  useEffect(() => {
+    value?.dispatch({
+      type: 'documentelementchangerequest',
+      detail: globalThis.document,
+    });
+    return () => {
+      value?.dispatch({
+        type: 'documentelementchangerequest',
+        detail: undefined,
+      });
+    };
+  }, []);
+  return (
+    <MediaContext.Provider value={value}>{children}</MediaContext.Provider>
+  );
+};
+
+export const useMediaStore = () => {
+  const store = useContext(MediaContext);
+  return store;
+};
+
+// These are hooks, which are the norm for working in "modern React". You can see them in action
+// in the PlayButton component and the Video component
+
+// This is a hook to get access to the media ui store's dispatch method, which allows the component to make
+// media state change requests. All it does is grab the context (with the "inherited" value provided from
+// the ancestor Provider) and return the dispatch. Simple.
+export const useMediaDispatch = () => {
+  const store = useContext(MediaContext);
+  const dispatch = store?.dispatch ?? console.log.bind(null, 'fake dispatch!');
```

Comment:
Just wondering what the scenarios are where the console log fallback will activate, maybe if they use the dispatch hook but haven't added the provider to their tree yet?

Reply:
Yeah I believe this was a timing issue? I def added it while testing. Good callout though I should at least make this less of a toy fallback.